docs/maintainer/example_recipes/go.md

```diff
@@ -0,0 +1,64 @@
+---
+title: 'Go packages'
+---
+
+If you want to package a Go package to conda-forge, you can use this recipe template:
+
+```yaml
+context:
+  name: example-package
+  version: "0.1.0"
+
+package:
+  name: ${{ name|lower }}
+  version: ${{ version }}
+
+source:
+  url: https://github.com/example-package/${{ name }}/archive/refs/tags/v${{ version }}.tar.gz
+  sha256: 0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef
+  target_directory: src
+
+build:
+  number: 0
+  script:
+    - cd src
+    - go-licenses save . --save_path ../library_licenses
+    - if: unix
+      then: go build -v -o $PREFIX/bin/saml-auth-proxy
+      else: go build -v -o %LIBRARY_BIN%\saml-auth-proxy.exe
+
+requirements:
+  build:
+    - ${{ compiler("go") }}
+    - go-licenses
+
+tests:
+  - script: example-package -help
+  - package_contents:
+    bin:
+      - example-package
+    strict: true
+
+about:
+  homepage: https://github.com/example-package/example-package
+  summary: Summary of the package.
+  description: |
+    Description of the package
+  license: MIT
+  license_file:
+    - src/LICENSE
+    - library_licenses/
+  documentation: https://pkg.go.dev/github.com/example-package/example-package
+  repository: https://github.com/example-package/example-package
+
+extra:
+  recipe-maintainers:
+    - LandoCalrissian
+```
+
+This recipe template supports different features:
+
+- Package licenses of statically linked libraries.
+- Ensure only binary is created by using `strict: true` in the `package_contents` tests.
```

Comment:
technically, this did not go through a CEP yet, right @wolfv?
Should we skip this for now?

Reply:
yep, we haven't ratified this yet ...

Reply:
It's important to note here that a significant number of Go packages also ship completions/man pages, so those will need to be allowed in the package as well, but I guess they can just be specified in the `package_contents` list as we well?